test/functional/wallet_hd.py

```diff
@@ -27,6 +29,57 @@ def set_test_params(self):
     def skip_test_if_missing_module(self):
         self.skip_if_no_wallet()
 
+    def test_addhdkey(self):
+        self.log.info("Test addhdkey")
+        def_wallet = self.nodes[0].get_wallet_rpc(self.default_wallet_name)
+        self.nodes[0].createwallet("hdkey")
+        wallet = self.nodes[0].get_wallet_rpc("hdkey")
+
+        assert_equal(len(wallet.gethdkeys()), 1)
+
+        wallet.addhdkey()
+        xpub_info = wallet.gethdkeys()
+        assert_equal(len(xpub_info), 2)
+        for x in xpub_info:
+            if len(x["descriptors"]) == 1 and x["descriptors"][0]["desc"].startswith("unused("):
+                break
+        else:
+            assert False, "Did not find HD key with no descriptors"
+
+        imp_xpub_info = def_wallet.gethdkeys(private=True)[0]
+        imp_xpub = imp_xpub_info["xpub"]
+        imp_xprv = imp_xpub_info["xprv"]
+
+        assert_raises_rpc_error(-5, "Extended public key (xpub) provided, but extended private key (xprv) is required", wallet.addhdkey, imp_xpub)
+        add_res = wallet.addhdkey(imp_xprv)
+        expected_void_desc = descsum_create(f"unused({imp_xpub})")
```

Comment:
In https://github.com/bitcoin/bitcoin/commit/0d24f0af7a3430dd8861d99390d902cde21f4275 "wallet: Add addhdkey RPC"

```diff
- expected_void_desc = descsum_create(f"unused({imp_xpub})")
+ expected_unused_desc = descsum_create(f"unused({imp_xpub})") 
```